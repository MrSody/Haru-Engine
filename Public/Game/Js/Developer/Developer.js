export default class Developer {
    constructor (tileSize) {
        this.tileSize = tileSize;
        this.lineSize = 0.4;
        this.lineColor = "#44414B";
    }

    drawGrid (ctx, cXnull, cYnull) {
        ctx.strokeStyle = this.lineColor;
        ctx.lineWidth = this.lineSize;

        // VERTICALES
        ctx.beginPath();
        ctx.moveTo((cXnull * this.tileSize), 0);
        ctx.lineTo((cXnull * this.tileSize), ctx.canvas.height);
        ctx.stroke();

        // HORIZONTALES
        ctx.beginPath();
        ctx.moveTo(0, ((cYnull - 0.5) * this.tileSize));
        ctx.lineTo(ctx.canvas.width, ((cYnull - 0.5) * this.tileSize));
        ctx.stroke();
    }

    drawCollision (ctx, cXnull, cYnull, mapCollision) {
        if (mapCollision[cYnull][cXnull] === 1) {
            ctx.fillStyle = "#FF0000";
            ctx.fillRect((cXnull * this.tileSize), ((cYnull - 0.5) * this.tileSize), 32, 32);
        }
    }
}