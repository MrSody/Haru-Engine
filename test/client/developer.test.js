/**
 * @jest-environment jsdom
 */

const { createCanvas, loadImage } = require('canvas')

const WORLD = require('../../server/core/engine/modules/entities/world/world.js').World;
const CONSTANT = require('./constants');

import { response } from 'express';
import Developer from '../../client/public/game/js/developer/developer';

const World = new WORLD();


test('drawGrid', async () =>{
    const clsDeveloper = new Developer(World.tileSize);
    const canvas = createCanvas(576, 352);
    const ctx = canvas.getContext('2d');

    for (let countY = 0; countY < CONSTANT.responseMap.collision.length; countY++) {
        for (let countX = 0; countX < CONSTANT.responseMap.collision[countY].length; countX++) {
            clsDeveloper.drawGrid(ctx, countX, countY);
        }
    }

    const dataURL = canvas.toDataURL();
    expect(dataURL).toMatchSnapshot();
});

test('drawCollision', async () =>{
    const clsDeveloper = new Developer(World.tileSize);
    const canvas = createCanvas(576, 352);
    const ctx = canvas.getContext('2d');

    for (let countY = 0; countY < CONSTANT.responseMap.collision.length; countY++) {
        for (let countX = 0; countX < CONSTANT.responseMap.collision[countY].length; countX++) {
            clsDeveloper.drawGrid(ctx, countX, countY);
            clsDeveloper.drawCollision(ctx, countX, countY, CONSTANT.responseMap.collision);
        }
    }

    const dataURL = canvas.toDataURL();
    expect(dataURL).toMatchSnapshot();
});

test('drawVisionNpc', async () =>{
    const clsDeveloper = new Developer(World.tileSize);
    const canvas = createCanvas(576, 352);
    const ctx = canvas.getContext('2d');

    let visionDistance = 2;
    let pos = {x: 12, y: 8};

    clsDeveloper.drawVisionNpc(ctx, visionDistance, pos);

    for (let countY = 0; countY < CONSTANT.responseMap.collision.length; countY++) {
        for (let countX = 0; countX < CONSTANT.responseMap.collision[countY].length; countX++) {
            clsDeveloper.drawGrid(ctx, countX, countY);
        }
    }

    const dataURL = canvas.toDataURL();
    expect(dataURL).toMatchSnapshot();
});
