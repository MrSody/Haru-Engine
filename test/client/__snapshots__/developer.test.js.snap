// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`drawCollision 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAkAAAAFgCAYAAABT3sadAAAABmJLR0QA/wD/AP+gvaeTAAAKM0lEQVR4nO3aT6tbeR3H8e8vSZumU3TwlqKMuhHcuXEhoosQ9GaRrNwpuHDl1mUfQ5firqtZ+BDspiAXCyNjcedKEGRAGNEGL/RPmjYncZEO8wzyK3xeryfwOc3vnNw3p6nlYr2pTpaL9ca+ffv27du3b//cRj3HAQB6EEAAQBwBBADEEUAAQBwBBADEEUAAQBwBBADEEUAAQBwBBADEEUAAQBwBBADEEUAAQBwBBADEEUAAQJx2OV8N1Wrosn6s8ekq7Nu3b9++ffv2z8cbIAAgz3Kx3vTctm/fvn379u3bPzdvgACAOAIIAIgjgACAOAIIAIgjgACAOAIIAIgjgACAOAIIAIgjgACAOAIIAIgjgACAOAIIAIgjgACAOAIIAIgjgACAOAIIAIgjgACAOAIIAIgjgACAOAIIAIgjgACAOAIIAIgjgACAOAIIAIgjgACAOAIIAIgjgACAOAIIAIgjgACAOO1yvhqq1dBl/Vjj01XYt2/fvn379u2fjzdAAECe5WK96blt3759+/bt27d/bt4AAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBx2uV8NVSrocv6scanq7Bv3759+/bt2z8fb4AAgDzLxXrTc9u+ffv27du3b//cvAECAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOK0y/lqqFZDl/VjjU9XYd++ffv27du3fz6TXsPA6QGcDsO4x/ZudJqd7fdd9rfjSbWquvP2bZf955MbNaqqr77Zddm/vjGtUVVdvN522X82ndWoqr7+6kWX/c9nd2rXYxjembRRu3589eiix/hysd5UVdm3n7r/td3r2e///McHPfZ//uOf3p8cj2W/7/7Hn1513X/49Em3/Tc3p9vk599+v/0qvwECAAIJIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOK0y/lqqFZDl/VjjU9XYd9+5v6kyzC8H/ZV0c+//Y77VeX7FzqbHvo8/7vR2B+g8P3bx2Pde/2yy/y/Z3dO9x90Mmmjdv346tFFj/HlYr2pqrJvP3X/3vbV7ONPrx702P/lj35y/7/TW9vkzz99/8Ptdvbw6ZNu99/O/We/036V3wABAIEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHHa5Xw1VKuhy/qxxqersG8/c3/SZfhL+6rozz993/1nP3a/qnrf//EmVTX64kY4s0O9+wKim1ZVNw99nv/dqMttx3vk1uFQd3evumz/59YHtW+tyzYn39ht2/f/96zLF8FfLu61Zzemxx7bX5i0Ubt+fPXoosf4crHeVFUl73/z5YvZw6d/etBj/1c/XNz/fHZ7m/z5997/yPlHn3/v/bvb7ezh0yfd7r+t+6/r/vRwmP3m73/rdv5t1Lqdf5XfAAEAgQQQABBHAAEAcQQQABBHAAEAcQQQABBHAAEAcQQQABBHAAEAcQQQABBHAAEAcQQQABBHAAEAcQQQABBHAAEAcQQQABBHAAEAcQQQABBHAAEAcQQQABBHAAEAcQQQABBHAAEAcQQQABBHAAEAcQQQABBHAAEAcQQQABBHAAEAcdrlfDVUq6HL+rHGp6vI3Z90Gf7Svir68++97/ztu/9yP//e+9HnX1W9//3x3osvALoZVdWNQ5/jfzMa17Sqbu73Xe6D3XhSb3oMv0fuHA71lbe7Lp//9c1bNT4M7eLN62OP/WfT2/Vi5D8hekr/+zNpo3b9+OrRRY/x5WK9qaqybz91/zvPr2e//esnD3rs//oH8/uzw7567n/2wZ1t8vl/6+Xz6PN/GX7+9vvtV/kNEAAQSAABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQRwABAHEEEAAQp13OV0O1GrqsH2t8ugr79jP3J12G3x/7KucfLP387Xfcr6r05w+68gcgm/OHjpaL9abntn379u3bt2/f/rn5DRAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEEcAAQBxBBAAEKddzldDtRq6rB9rfLoK+/bt27dv37798/EGCADIs1ysNz237du3b9++ffv2z80bIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgjgACAOIIIAAgTrucr4ZqNXRZP9b4dBX27du3b9++ffvn4w0QAJBnuVhvem7bt2/fvn379u2fmzdAAEAcAQQAxBFAAEAcAQQAxBFAAEAcAQQAxBFAAEAcAQQAxBFAAEAcAQQAxBFAAEAcAQQAxBFAAEAcAQQAxBFAAEAcAQQAxBFAAEAcAQQAxBFAAEAcAQQAxBFAAEAcAQQAxBFAAEAcAQQAxBFAAEAcAQQAxBFAAEAcAQQAxBFAAECcdjlfDdVq6LJ+rPHpKuzbt2/fvn379s9n0msY4MP929bq3Rfhme1ba9PDUL3237ZRu57cOPbYBqombdSuH189uugxvlysN1VV9u3bz9z/2b/+OfvFZ/940GP/d9/93v27u2313P/DR9/eJp+/ffu99qv8BggACCSAAIA4AggAiCOAAIA4AggAiCOAAIA4AggAiCOAAIA4AggAiCOAAIA4AggAiCOAAIA4AggAiCOAAIA4/weyOiqgQtKmkAAAAABJRU5ErkJggg=="`;

exports[`drawGrid 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAkAAAAFgCAYAAABT3sadAAAABmJLR0QA/wD/AP+gvaeTAAAHkElEQVR4nO3YsZGDQBBFweUUmApDGJB/GIRA6QzlsGu87gT+mK9mHPt1j0WO/brt27dv3759+/Zn+1s5DgCwggACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgZ/u8z2ds41my/h2v3xX27du3b9++ffvz+AABAD3Hft0rt+3bt2/fvn379mfzAQIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgZ/u8z2ds41my/h2v3xX27du3b9++ffvz+AABAD3Hft0rt+3bt2/fvn379mfzAQIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgZ/u8z2ds41my/h2v3xX27du3b9++ffvz+AABAD3Hft0rt+3bt2/fvn379mfzAQIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgZ/u8z2ds41my/h2v3xX27du3b9++ffvz+AABAD3Hft0rt+3bt2/fvn379mfzAQIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgZ/u8z2ds41my/h2v3xX27du3b9++ffvz+AABAD3Hft0rt+3bt2/fvn379mfzAQIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgZ/u8z2ds41my/h2v3xX27du3b9++ffvz+AABAD3Hft0rt+3bt2/fvn379mfzAQIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgZ/u8z2ds41my/h2v3xX27du3b9++ffvz+AABAD3Hft0rt+3bt2/fvn379mfzAQIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgZ/u8z2ds41my/h2v3xX27du3b9++ffvz+AABAD3Hft0rt+3bt2/fvn379mfzAQIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgZ/u8z2ds41my/h2v3xX27du3b9++ffvz+AABAD3Hft0rt+3bt2/fvn379mfzAQIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgZ/u8z2ds41my/h2v3xX27du3b9++ffvz+AABAD3Hft0rt+3bt2/fvn379mfzAQIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgZ/u8z2ds41my/h2v3xX27du3b9++ffvz+AABAD3Hft0rt+3bt2/fvn379mfzAQIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOf9uo3TZTqIOfgAAAABJRU5ErkJggg=="`;

exports[`drawVisionNpc 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAkAAAAFgCAYAAABT3sadAAAABmJLR0QA/wD/AP+gvaeTAAAIpklEQVR4nO3aMW5TWxSG0X3jpKBmQJGFYgl7QIyBAQESEmJElugoootf4TcGn+JbawL/aW7yactzOl6us8jpeLnat2/fvn379u0/2tPKcQCAFQQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOdvb63mfbfYl67c53F9h3759+/bt27f/OC5AAEDP6Xi5rty2b9++ffv27dt/NBcgACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHK2t9fzPtvsS9Zvc7i/wr59+/bt27dv/3FcgACAntPxcl25bd++ffv27du3/2guQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkbG+v53222Zes3+Zwf4V9+/bt27dv3/7juAABAD2n4+W6ctu+ffv27du3b//RXIAAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyNneXs/7bLMvWb/N4f4K+/bt27dv3779x3EBAgB6TsfLdeW2ffv27du3b9/+o7kAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJCzvb2e99lmX7J+m8P9Ffbt27dv3759+4/jAgQA9JyOl+vKbfv27du3b9++/UdzAQIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgZ3t7Pe+zzb5k/TaH+yvs27dv3759+/Yf53nVMMByLzMz//8hXsd+df992TIz87w9bX9+/v72ccX46Xi5zszYt2/f/or9/eXfh1/ff3xdsf/p/PnLzIz97v5hf/pb/v5W7s/4DRAAECSAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOdvb63mfbfYl67c53F9h3759+wv2X5Ysw937tL+/lfvjAgQABD1vT9ufn7+/fVwxfjperjMz9u3bt79if3/59+HX9x9fV+x/On/+MjNjv7t/2J/+lr+/lfszLkAAQJAAAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5Gxvr+d9ttmXrN/mcH+Fffv27S/Yf1myDHfv0/7+Vu6PCxAAEPS8PW1/fv7+9nHF+Ol4uc7M2Ldv3/6K/f3l34df3398XbH/6fz5y8yM/e7+YX/6W/7+Vu7PuAABAEECCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkLO9vZ732WZfsn6bw/0V9u3bt79g/2XJMty9T/v7W7k/LkAAQNDz9rT9+fn728cV46fj5TozY9++ffsr9veXfx9+ff/xdcX+p/PnLzMz9rv7h/3pb/n7W7k/4wIEAAQJIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgRwABADkCCADIEUAAQM729nreZ5t9yfptDvdX2Ldv3/6C/Zcly3D3Pu3vb+X+uAABAEHP29P25+fvbx9XjJ+Ol+vMjH379u2v2N9f/n349f3H1xX7n86fv8zM2O/uH/anv+Xvb+X+jAsQABAkgACAHAEEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHIEEACQI4AAgBwBBADkCCAAIEcAAQA5AggAyBFAAECOAAIAcgQQAJAjgACAHAEEAOQIIAAgRwABADnb2+t5n232Jeu3OdxfYd++ffsL9l+WLMPd+7S/v5X7M/O8ahhgOf+A7K/eZ53T8XJduW3fvn379u3bt/9ofgMEAOQIIAAgRwABADkCCADIEUAAQI4AAgByBBAAkCOAAIAcAQQA5AggACBHAAEAOQIIAMgRQABAjgACAHL+A+Xzmn2Zb0vJAAAAAElFTkSuQmCC"`;
